apps/src/sites/code.org/pages/views/workshop_search.js

```diff
@@ -10,30 +11,265 @@ const markerCustererOptions = {
+    layout: {
+      'text-field': ['get', 'clustered_workshop_count'],
+      'text-font': ['DIN Offc Pro Medium', 'Arial Unicode MS Bold'],
+      'text-size': 12,
+      'text-allow-overlap': true
+    }
+  });
+  map.on('click', 'workshop-clusters', function(e) {
+    var features = map.queryRenderedFeatures(e.point, {
+      layers: ['workshop-clusters']
+    });
+    var clusterId = features[0].properties.cluster_id;
+    map
+      .getSource('workshops')
+      .getClusterExpansionZoom(clusterId, function(err, zoom) {
+        if (err) {
+          console.log(err);
+          return;
+        }
+
+        map.easeTo({
+          center: features[0].geometry.coordinates,
+          zoom: zoom
+        });
+      });
+  });
+  map.on('mouseenter', 'workshop-clusters', function() {
+    map.getCanvas().style.cursor = 'pointer';
+  });
+}
+
+function placeIntroWorkshops() {
+  map.loadImage(
+    'https://maps.google.com/mapfiles/kml/paddle/red-blank.png',
+    (error, image) => {
+      if (error) {
+        console.log(error);
+        return;
+      }
+      map.addImage('marker', image);
+      map.addLayer({
+        id: 'intro-workshops',
+        type: 'symbol',
+        source: 'workshops',
+        filter: [
+          'all',
+          ['==', 'show_deep_dive_marker', 'false'],
+          ['!has', 'point_count']
+        ],
+        layout: {
+          'icon-image': 'marker',
+          'icon-size': 0.5,
+          'icon-anchor': 'bottom',
+          'icon-allow-overlap': true
+        }
+      });
+    }
+  );
+  map.on('click', 'intro-workshops', onMarkerClick);
+  map.on('mouseenter', 'intro-workshops', function() {
+    map.getCanvas().style.cursor = 'pointer';
+  });
+}
+
+function placeDeepDiveWorkshops() {
+  map.loadImage(
+    'https://maps.google.com/mapfiles/kml/paddle/red-stars.png',
+    (error, image) => {
+      if (error) {
+        console.log(error);
+        return;
+      }
+      map.addImage('star-marker', image);
+      map.addLayer({
+        id: 'deep-dive-workshops',
+        type: 'symbol',
+        source: 'workshops',
+        filter: [
+          'all',
+          ['==', 'show_deep_dive_marker', 'true'],
+          ['!has', 'point_count']
+        ],
+        layout: {
+          'icon-image': 'star-marker',
+          'icon-size': 0.5,
+          'icon-anchor': 'bottom',
+          'icon-allow-overlap': true
+        }
+      });
+    }
+  );
+  map.on('click', 'deep-dive-workshops', onMarkerClick);
+  map.on('mouseenter', 'deep-dive-workshops', function() {
+    map.getCanvas().style.cursor = 'pointer';
+  });
+}
+
+function onMarkerClick(e) {
+  var coordinates = e.features[0].geometry.coordinates.slice();
+  var workshops = JSON.parse(e.features[0].properties['workshops']);
+  const description = compileMapboxPopupHtml(workshops, false);
+
+  new mapboxgl.Popup()
+    .setLngLat(coordinates)
+    .setHTML(description)
+    .addTo(map);
+}
+
+function compileMapboxPopupHtml(workshops, first) {
```

Comment:
This is mostly copied from the Google Maps version but the data is structured slightly differently. Ideally we would deduplicate this code but we should be removing the google maps code soon.